repose-aggregator/components/filters/scripting-filter/src/test/scala/org/openrepose/filters/scripting/ScriptingFilterTest.scala

```diff
@@ -0,0 +1,223 @@
+    val filterChain = new MockFilterChain()
+
+    val scriptingConfig = new ScriptingConfig()
+
+    val scriptData = new ScriptData()
+    scriptData.setValue(
+      """
+        |request.addHeader("lol", "butts")
+        |filterChain.doFilter(request, response)
+      """.stripMargin)
+    scriptData.setLanguage("python")
+    scriptingConfig.setScript(scriptData)
+
+    filter.configurationUpdated(scriptingConfig)
+
+    filter.doFilter(new MockHttpServletRequest(), new MockHttpServletResponse(), filterChain)
+    filterChain.getRequest.asInstanceOf[HttpServletRequest].getHeader("lol") should equal("butts")
+  }
+
+  it("can parse some lua to add a header with static value") {
+    val fakeConfigService = new FakeConfigService()
+    val filter = new ScriptingFilter(fakeConfigService)
+    val filterChain = new MockFilterChain()
+
+    val scriptingConfig = new ScriptingConfig()
+
+    val scriptData = new ScriptData()
+    scriptData.setValue(
+      """
+        |request:addHeader("lol", "butts")
+        |filterChain:doFilter(request, response)
+      """.stripMargin)
+    scriptData.setLanguage("lua")
+    scriptingConfig.setScript(scriptData)
+
+    filter.configurationUpdated(scriptingConfig)
+
+    filter.doFilter(new MockHttpServletRequest(), new MockHttpServletResponse(), filterChain)
+    filterChain.getRequest.asInstanceOf[HttpServletRequest].getHeader("lol") should equal("butts")
+  }
+
+  ignore("can parse some scala to add a header with static value") {
+    val fakeConfigService = new FakeConfigService()
+    val filter = new ScriptingFilter(fakeConfigService)
+    val filterChain = new MockFilterChain()
+
+    val scriptingConfig = new ScriptingConfig()
+
+    val scriptData = new ScriptData()
+    scriptData.setValue(
+      """
+        |request.addHeader("lol", "butts")
+        |filterChain.doFilter(request, response)
+      """.stripMargin)
+    scriptData.setLanguage("scala")
+    scriptingConfig.setScript(scriptData)
+
+    filter.configurationUpdated(scriptingConfig)
+
+    filter.doFilter(new MockHttpServletRequest(), new MockHttpServletResponse(), filterChain)
+    filterChain.getRequest.asInstanceOf[HttpServletRequest].getHeader("lol") should equal("butts")
+  }
+
+  it("should return a 500 if the script throws an exception") {
+    val filter = new ScriptingFilter(null)
+    val filterChain = new MockFilterChain()
+
+    val scriptingConfig = new ScriptingConfig()
+
+    val scriptData = new ScriptData()
+    scriptData.setValue(
+      """
+        |throw "EXCEPTION"
+      """.stripMargin)
+    scriptData.setLanguage("javascript")
+    scriptingConfig.setScript(scriptData)
+
+    filter.configurationUpdated(scriptingConfig)
+
+    val request = new MockHttpServletRequest()
+    val response = new MockHttpServletResponse()
+
+    filter.doFilter(request, response, filterChain)
+
+    filterChain.getRequest shouldBe null
+    response.getStatus shouldBe HttpServletResponse.SC_INTERNAL_SERVER_ERROR
+  }
+}
```

Comment:
I don't think I saw any tests for adding a header in the response.  Other than that, I love seeing the various languages that we know will work.